src/compute/src/render/context.rs

```diff
@@ -303,6 +310,26 @@ impl HydrationLogger {
     }
 }
 
+/// A logger for [`ComputeEvent`]s emitted for a dataflow.
+#[derive(Clone)]
+pub(super) struct ComputeEventLogger {
+    export_ids: Vec<GlobalId>,
+    logger: crate::logging::compute::Logger,
+}
+
+impl ComputeEventLogger {
+    /// Log the given compute events.
+    pub fn log_many<'a>(&self, events: impl Iterator<Item = &'a ComputeEvent>) {
+        self.logger.log_many(events);
+    }
+
+    /// Log a compute event for each dataflow export.
+    pub fn log_for_exports(&self, f: impl Fn(GlobalId) -> ComputeEvent) {
+        let events: Vec<_> = self.export_ids.iter().map(|id| f(*id)).collect();
```

Comment:
This doesn't need to allocate, doesn't it?

Reply:
Probably not, but I can't figure out the types! `log_many` wants an iterator over references but here we get an iterator over owned types, and to convert that into an iterator over references, we need to stash the owned types somewhere. Maybe `log_many` could be made to work with owned types somehow?

Reply:
Ah, I see. I think you can change the signature for `log_many` to mirror that of the underlying logger, which is generic over some `PushInto`.